### DISCUSSION TITLE
Is it possible to change the indent distance of list items?

### DISCUSSION THREAD

**Discussion by ylin** [gonzalezreal/swift-markdown-ui#322] [Q&A]
I find the left indent of lists to be too wide. Here is what it looks like now:

![CleanShot 2024-06-07 at 10 25 46@2x](https://github.com/gonzalezreal/swift-markdown-ui/assets/312667/e57e706d-a12b-40e7-a09b-eea712382a56)

I would like it to look like this:

![CleanShot 2024-06-07 at 10 26 12@2x](https://github.com/gonzalezreal/swift-markdown-ui/assets/312667/2e7e86d1-2480-421a-9346-d5de456240cc)

If you can point me to the code, perhaps I can create a PR for it. Best if it's already built in though, but for .markdownMargin(), I only see the top and bottom options, not leading.

**Comment by gonzalezreal** [comment]
Hi @ylin,

Currently, the list markers have a minimum width of 1.5em (that is 1.5 * current font size):
https://github.com/gonzalezreal/swift-markdown-ui/blob/c0daf6eb79d97964180f3113868c990bd1c4a007/Sources/MarkdownUI/Theme/BlockStyle/ListMarkerConfiguration.swift#L21

You can create custom list markers with a different minimum width (i.e. 1em) to achieve the desired effect.

**Comment by EryouHao** [comment]
@gonzalezreal Hi, Can you tell me how to adjust this setting in markdownBlockStyle?

I seem to have not found the correct method.

```swift
import SwiftUI
import MarkdownUI

struct MarkdownView: View {
    let note: Note
    let containerHeight: CGFloat = 180
    
    var body: some View {
        Markdown(note.content)
            .markdownTheme(.docC)
            .markdownTextStyle {
                FontSize(10)
            }
            .markdownBlockStyle(\.listItem) { configuration in
                configuration.label.relativeFrame(minWidth: .em(0.5), alignment: .trailing) // ✖ not work
            }
            .frame(maxHeight: containerHeight, alignment: .top)
//          .clipped()
            .mask {
                  LinearGradient(
                    colors: [.black, .clear],
                    startPoint: UnitPoint(x: 0.5, y: 1 - 10 / containerHeight), endPoint: .bottom
                  )
                }
    }
}
```

**Reply by gonzalezreal** [reply]
You need to override the `bulletedListMarker`, `numberedListMarker` and `taskListMarker` styles instead of the `listItem` one.

For instance, this replaces the bulleted list markers with a dash `-` using the current font size (1em) as the indent size.

```swift
Markdown("...")
  .markdownBlockStyle(\.bulletedListMarker) { configuration in
    Text("-")
      .relativeFrame(minWidth: .em(1), alignment: .trailing)
  }
```

**Reply by EryouHao** [reply]
@gonzalezreal Thanks for your reply. but it seems still not work.

<img width="1193" alt="image" src="https://github.com/gonzalezreal/swift-markdown-ui/assets/17328747/53a9fd42-cd73-45a6-afd4-2b27aaeb917b">


my code:
```swift
import SwiftUI
import MarkdownUI

struct MarkdownView: View {
    let note: Note
    let maxHeight: CGFloat?
    let fontSize: CGFloat
    
    init(note: Note, fontSize: CGFloat = 14, maxHeight: CGFloat? = nil) {
        self.note = note
        self.fontSize = fontSize
        self.maxHeight = maxHeight
    }
    
    var body: some View {
        let content = Markdown(note.content)
            .markdownTheme(.docC)
            .markdownTextStyle {
                FontSize(fontSize)
            }
            .markdownBlockStyle(\.bulletedListMarker) { configuration in
                Text("-")
                  .relativeFrame(minWidth: .em(1), alignment: .trailing)
              }
        
        if let maxHeight = maxHeight {
            return AnyView(content.frame(maxHeight: maxHeight, alignment: .top).mask {
                LinearGradient(
                    colors: [.black, .clear],
                    startPoint: UnitPoint(x: 0.5, y: 1 - 10 / maxHeight), endPoint: .bottom
                )
            })
        } else {
            return AnyView(content)
        }
    }
}

```

**Reply by EryouHao** [reply]
Got it. because I use `.docC` theme, when I remove it, it works.

<img width="1097" alt="image" src="https://github.com/gonzalezreal/swift-markdown-ui/assets/17328747/7826caf8-1eeb-4d3b-a084-3e2e875f5dc2">
